.github/workflows/test/summarize-checks.test.js

```diff
@@ -511,4 +512,105 @@ describe("summarizeChecksImpl", () => {
+        expectedLabelsToAdd: [],
+        expectedLabelsToRemove: ["WaitForARMFeedback", "ARMChangesRequested"]
+      },
+      {
+        description: "labeled: ARMSignedOff with only WaitForARMFeedback present",
+        eventName: "labeled",
+        changedLabel: "ARMSignedOff",
+        existingLabels: ["WaitForARMFeedback", "ARMSignedOff", "other-label"],
+        expectedLabelsToAdd: [],
+        expectedLabelsToRemove: ["WaitForARMFeedback"]
+      },
+      {
+        description: "labeled: ARMSignedOff with only ARMChangesRequested present",
+        eventName: "labeled",
+        changedLabel: "ARMSignedOff",
+        existingLabels: ["ARMChangesRequested", "ARMSignedOff", "other-label"],
+        expectedLabelsToAdd: [],
+        expectedLabelsToRemove: ["ARMChangesRequested"]
+      },
+      {
+        description: "labeled: ARMSignedOff with neither label present",
+        eventName: "labeled",
+        changedLabel: "ARMSignedOff",
+        existingLabels: ["other-label", "ARMSignedOff"],
+        expectedLabelsToAdd: [],
+        expectedLabelsToRemove: []
+      },
+      {
+        description: "unlabeled: ARMChangesRequested with WaitForARMFeedback present (buggy case)",
+        eventName: "unlabeled",
+        changedLabel: "ARMChangesRequested",
+        existingLabels: ["WaitForARMFeedback", "other-label"],
+        expectedLabelsToAdd: [],
+        expectedLabelsToRemove: []
+      },
+      {
+        description: "unlabeled: ARMChangesRequested without WaitForARMFeedback",
+        eventName: "unlabeled",
+        changedLabel: "ARMChangesRequested",
+        existingLabels: ["other-label"],
+        expectedLabelsToAdd: ["WaitForARMFeedback"],
+        expectedLabelsToRemove: []
+      },
+      {
+        description: "labeled: other label should have no effect",
+        eventName: "labeled",
+        changedLabel: "SomeOtherLabel",
+        existingLabels: ["WaitForARMFeedback", "ARMChangesRequested"],
+        expectedLabelsToAdd: [],
+        expectedLabelsToRemove: ["WaitForARMFeedback"]
+      },
+      {
+        description: "unlabeled: other label should have no effect",
+        eventName: "unlabeled",
+        changedLabel: "SomeOtherLabel",
+        existingLabels: ["WaitForARMFeedback", "ARMChangesRequested"],
+        expectedLabelsToAdd: [],
+        expectedLabelsToRemove: ["WaitForARMFeedback"]
+      }
+    ];
+
+    it.each(testCases)(
+      "$description",
+      async ({ eventName, changedLabel, existingLabels, expectedLabelsToAdd, expectedLabelsToRemove }) => {
+        const [labelsToAdd, labelsToRemove] = await updateLabels(
+            eventName,
+            "main",
+            existingLabels,
+            changedLabel
+        );
+
+        expect(labelsToAdd.sort()).toEqual(expectedLabelsToAdd.sort());
+        expect(labelsToRemove.sort()).toEqual(expectedLabelsToRemove.sort());
+      },
+      600000,
```

Comment:
Why do we need to set the timeout?  Shouldn't these be fast?

Reply:
I will clean this up. This is more of a local integration test.